playbooks/upload_to_swift.yml

```diff
@@ -33,24 +80,28 @@
         object_cdn: "{{ service_catalog | selectattr('type', 'equalto', 'rax:object-cdn') | first }}"
       when: rax_pub_cloud
 
-    # os_object does not currently support setting object expiration header field
-    - name: Upload file to Cloud Files
-      command: "swift upload {{ container }} {{ archive_name }} --header 'X-Delete-After:2592000'"
-      args:
-        chdir: "{{ artifacts_dir | dirname }}"
-      no_log: true
-      environment:
-        OS_AUTH_TOKEN: "{{ auth_token }}"
-        OS_STORAGE_URL: "{{ (object_store['endpoints'] | selectattr('region', 'equalto', region) | first)['publicURL'] }}"
+    - set_fact:
+        object_cdn_url: "{{ (object_cdn['endpoints'] | selectattr('region', 'equalto', region) | first)['publicURL'] }}/{{ container }}"
+
+    - set_fact:
+        object_store_url: "{{ (object_store['endpoints'] | selectattr('region', 'equalto', region ) | first)['publicURL']}}"
+
+    - name: Enable CDN
+      uri:
+        url: "{{object_cdn_url}}"
+        method: PUT
+        headers:
+          X-AUTH-TOKEN: "{{ auth_token }}"
+          X-Cdn-Enabled: True
+        status_code: "200, 201, 202, 204"
 
     - name: Get Rackspace CloudFiles CDN URL
       uri:
-        url: "{{ (object_cdn['endpoints'] | selectattr('region', 'equalto', region) | first)['publicURL'] }}/{{ container }}"
+        url: "{{ object_cdn_url }}"
         method: HEAD
         headers:
           X-AUTH-TOKEN: "{{ auth_token }}"
         status_code: 204
-      no_log: true
```

Comment:
Are we removing no_log for a reason? It's there to prevent the leak of the token in any output.

Reply:
That was debugging, I've now reinstated it, and added to other similar tasks.